src/components/layout/Snackbar.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import { observer } from 'mobx-react'
+import { Snackbar } from '@material-ui/core'
+import { Alert } from '@material-ui/lab'
+import { AlertStore } from '../../stores/AlertStore'
+
+function SnackBar() {
+  const { getAlerts } = AlertStore
+
+  const handleClose = () => AlertStore.hide()
+  const handleExited = () => AlertStore.clear()
+
+  return (
+    <>
+      {getAlerts.map(({ id, show, duration, type, message }) => {
+        return (
+          <Snackbar
+            key={id}
+            open={show}
+            autoHideDuration={duration}
+            onClose={handleClose}
+            onExited={handleExited}>
+            <Alert severity={type} onClose={handleClose}>
```

Comment:
At the moment we tend to hide all alerts on click via the [`clickaway` listener](https://material-ui.com/api/snackbar/#props) of `<Snackbar />`

We might discard the clickaway hiding by checking the `reason: SnackbarCloseReason` of the handler.

Another question: Do you think each individual alert should clear all alerts when it exits it's transition via `onExited`?

How do you find such approach @StanislavSt ?

```tsx
import React from 'react'
import { observer } from 'mobx-react'
import { Snackbar, SnackbarCloseReason } from '@material-ui/core'
import { Alert } from '@material-ui/lab'
import { AlertStore } from 'stores/AlertStore'

function SnackBar() {
  const { getAlerts } = AlertStore

  const handleSnackBarClose = (id: number) => (
    _event: React.SyntheticEvent,
    reason: SnackbarCloseReason,
  ) => {
    if (reason === 'clickaway') {
      return
    }
    AlertStore.hide(id)
  }
  const handleClose = (id: number) => () => AlertStore.hide(id)

  return (
    <>
      {getAlerts.map(({ id, show, duration, type, message }) => {
        return (
          <Snackbar
            key={id}
            open={show}
            autoHideDuration={duration}
            onClose={handleSnackBarClose(id)}>
            <Alert severity={type} onClose={handleClose(id)}>
              {message}
            </Alert>
          </Snackbar>
        )
      })}
    </>
  )
}

export default observer(SnackBar)
```

Reply:
@kachar  Thank you for the fast review! 

I agree that we should not close the alert on `clickaway`.
I agree that `onExited` is not needed, I used it for an animation problem - but now when I tested again it is not needed.

Using the `id` approach will cause problems, such as after the `autoHideDuration` has expired you will close the current  alert. For example if I show multiple alerts one after another, the last one might close much earlier before the 3000ms has expired, because a callback comes from a previous alert. 

Using `id = this.alerts.length` to hide an alert
![Alert_using_id](https://user-images.githubusercontent.com/12186099/107053365-0386f700-67cf-11eb-915d-d9e4ebd125d6.gif)

Without using  id to hide an alert:
![alert_without_id](https://user-images.githubusercontent.com/12186099/107053432-17325d80-67cf-11eb-8c03-e0314bd8043b.gif)


We could use an unique id and hide based on that, something like this:
```
 hide(id: number) {
    const index = this.alerts.findIndex((alert) => alert.id === id)
    this.alerts[index].show = false
  }
```
But I think the result will be the same, since we always clear the Alerts array , before pushing a new one.

Reply:
I think we're good to go with this one.
You're right about the multiple alerts storage handling in the case we show a single alert at all times.

If the expectations for the behavior evolve we'll update the code accordingly.

Thanks again for the energy well spent on this PR! :zap: